tests/Doctrine/Tests/ORM/Functional/ParserResultSerializationTest.php

```diff
@@ -33,12 +34,23 @@ public function testSerializeParserResult(): void
 
         $parserResult = self::parseQuery($query);
         $serialized   = serialize($parserResult);
+        $this->assertStringNotContainsString(
+            '_sqlStatements',
+            $serialized,
+            'ParserResult should not contain any reference to that legacy property'
+        );
         $unserialized = unserialize($serialized);
 
         $this->assertInstanceOf(ParserResult::class, $unserialized);
         $this->assertInstanceOf(ResultSetMapping::class, $unserialized->getResultSetMapping());
         $this->assertEquals(['name' => [0]], $unserialized->getParameterMappings());
         $this->assertInstanceOf(SingleSelectExecutor::class, $unserialized->getSqlExecutor());
+        $this->assertIsString($unserialized->getSqlExecutor()->getSqlStatements());
+
+        $r = new ReflectionProperty($unserialized->getSqlExecutor(), '_sqlStatements');
+        $r->setAccessible(true);
+
+        $this->assertIsString($r->getValue($unserialized->getSqlExecutor()));
```

Comment:
Even though `_sqlStatements` is referencing `sqlStatements` this test would be more understandable if you compare the contents of both properties.

```suggestion
        $this->assertSame($r->getValue($unserialized->getSqlExecutor()), $unserialized->getSqlExecutor()->getSqlStatements());
```

Maybe you want to create a separated test method for your changes instead of re-using `testSerializeParserResult()`. One tests the serialized ParserResult, the other your forward compatiility.